uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1547,69 +1573,75 @@ sub CreateMRICandidates {
-            $sex = "Female";
+        $sex = "Female";
     } elsif ($tarchiveInfo->{'PatientSex'} eq 'M') {
         $sex = "Male";
     }
 
-    ################################################################
-    ## Create non-existent candidate if the profile allows for #####
-    ## Candidate creation ##########################################
-    ################################################################
-    if (!NeuroDB::MRI::subjectIDExists(
-            $subjectIDsref->{'CandID'},
-            $this->{dbhr}
-        ) 
-        && (NeuroDB::DBI::getConfigSetting($this->{dbhr},'createCandidates'))
-    ) {
-           chomp($User);
-            unless ($subjectIDsref->{'CandID'}) {
-                $subjectIDsref->{'CandID'} = 
-                NeuroDB::MRI::createNewCandID($this->{dbhr});
-            }
-            $query = "INSERT INTO candidate ".
-                     "(CandID, PSCID, DoB, Sex, RegistrationCenterID, ".
-                     "Date_active, Date_registered, UserID, Entity_type) ".
-                     "VALUES(" . 
-                     ${$this->{'dbhr'}}->quote($subjectIDsref->{'CandID'}).",".
-                     ${$this->{'dbhr'}}->quote($subjectIDsref->{'PSCID'}).",".
-                     ${$this->{'dbhr'}}->quote($tarchiveInfo->{'PatientDoB'}) ."," .
-                     ${$this->{'dbhr'}}->quote($sex).",".
-                     ${$this->{'dbhr'}}->quote($centerID). 
-                     ", NOW(), NOW(), '$User', 'Human')";
-            
-            if ($this->{debug}) {
-                print $query . "\n";
-            }
-            ${$this->{'dbhr'}}->do($query);
-            $message = "\n==> CREATED NEW CANDIDATE :
-            		$subjectIDsref->{'CandID'}";
-            $this->{LOG}->print($message);
-            $this->spool($message, 'N', $upload_id, $notify_detailed);
-      } elsif ($subjectIDsref->{'CandID'}) {# if the candidate exists
-            $message = "\n==> getSubjectIDs returned this CandID/DCCID : ".
-               "$subjectIDsref->{'CandID'}\n";
-	    $this->{LOG}->print($message);
-            $this->spool($message, 'N', $upload_id, $notify_detailed);
-      } else {
-            $message = "\nERROR: The candidate could not be considered for ". 
-                       "uploading, since s/he is not registered in your database.".
-                       "\nThe dicom header PatientID is: ". 
-                       $tarchiveInfo->{'PatientID'}. "\n ".
-                       "The dicom header PatientName is: ". 
-                       $tarchiveInfo->{'PatientName'}. "\n\n";
-            $this->writeErrorLog(
-                $message, $NeuroDB::ExitCodes::INSERT_FAILURE
-            );
-            $this->spool($message, 'Y', $upload_id, $notify_notsummary);
-            exit $NeuroDB::ExitCodes::INSERT_FAILURE;
-     }
+    chomp($User);
+    $candID = NeuroDB::MRI::createNewCandID($dbh) unless $candID;
+    $query  = "INSERT INTO candidate ".
```

Comment:
I think it'd be better if the query was prepared (with argument placeholders) and then executed with the appropriate arguments instead of building a "one-shot" query in which all the arguments are quoted.